### DISCUSSION TITLE
Unrealistic passive sonar - it should pick up noise

### DISCUSSION THREAD

**Discussion by Kulik242** [FakeFishGames/Barotrauma#4722] [Ideas and suggestions]
The purpose of passive sonar is to detect the acoustic signals emanating from external sources by listening.

In Barotrauma it seams, passive sonar has a set range at which it picks up all objects without regards to noise they emitting.
This simplification is ok for nonliving objects like the cavern walls, wreck hulls, ruins, etc., in deed, IRL these would be seen on the sonar screen as a reflection of submaerine's own noise (although noisy submarines should thus see farther).

On the other hand, in Barotrauma, the passive sonar doesn't picks up external noise sources like it should do. For instance, in real life, all aquatic vertebrae emit whole spectrum of noises which are easily picked by a passive sonar at long distances. This isn't represented in the game, Barotrauma's passive sonar picks up creatures only at the same range as inert objects.

I propose that the passive sonar should be reworked that noise sources should be clearly visible on the passive sonar - vertebrae both creatures and human, scooters, plasma cutting and welding, using crowbars, gunshots, explosions should be visible based on their noise intensity. Even a diver talking should be more visible than a diver that is silent- shouting works better underwater than above. Loud noises like gunshots and especially explosions should cause noise waves that would "illuminate" inert objects on passive sonar (explosions better than the active sonar ping). Active handheld sonar should also "illuminate" passive sonar view on the sub's passive sonar screen.

**Comment by Regalis11** [comment]
>In Barotrauma it seams, passive sonar has a set range at which it picks up all objects without regards to noise they emitting.

This isn't actually true, the way the passive sonar works in the game is that it picks up objects that are emitting sounds (monsters, devices, characters moving inside the sub, etc). It also detects _sounds reflected from static obstacles_, so for example, if a loud machine like the reactor or the engine is running, you're able to see some of the static walls outside the sub.

So what you described is very close to how it works already, although there are a lot of sources of sounds (handheld tools for instance) that aren't registered on the sonar. Changing this would have major implications on gameplay though, since monsters also use sounds to detect targets. But perhaps something to look into in the future.

**Reply by Saevon** [reply]
If its tool use outside a vehicle / base / wreck… maybe it should make noise?  Though balance wise we'd need way more ways to detect approaching enemies and stealth around them. (Or discourage them long enough to escape / hide)

I would love a very clear "this creature uses sound", "this creature uses light", "this creature is very sensitive to sound" kind of lore in the game...

**Reply by Ek-Videogames** [reply]
_works as 'intended' but not exactly well, when using passive you are essentially blind still._
Things that should fit your definition of 'making noise' still never seem to appear on sonar until they're basically against the hull of the sub itself.

A fascinating specific microcosm of the passive sonar issue is in PVP, where attempting to use stealth with passive is just a straight disadvantage against anyone who's using active. This being because the enemy's active sonar pings are not making them visible on your sonar at all, yet you on the other hand - because the enemy is on active - are showing up clearly on their sonar at the longest range.

(With monsters it is more or less the problem of having zero time to react **when** monsters appear, rather than ability to see and respond to them if you had been using active sonar)

**Reply by Saevon** [reply]
> Things that should fit your definition of 'making noise' still never seem to appear on sonar until they're basically against the hull of the sub itself.

I assume you mean in real life?     My "should" is actually gameplay-design wise, I think having to be careful in wrecks could be awesome.

> This being because the enemy's active sonar pings are not making them visible on your sonar at all,

Yeah they totally should be showing up!  "Ping" from that direction! 

But at the same time this means if anyone is using active, everyone is basically using active… no real way to hide.    Instead we would need "stealth profiles"   letting the sub be invisible to sonar… unless its using active sonar, engine noise, other noise.   Perhaps the stealth profile disappears at very close range (where the enemy sub starts seeing you on passive sonar.

This would mean that active sonar is used to detect terrain and navigate around a PvP map (assuming we get good map designs) and Passive is used to actually sneak up on an enemy (but you'll be lost in navigation, only seeing nearby walls and paths)

**Reply by Krzeszny** [reply]
The passive sonar isn't even usable. The switch works like Active/Off. In PvP it's a disadvantage to use it and the range is too low to even use it as a navigation device. You turn it to passive to save power and use no sonar.

**Reply by mLuby** [reply]
> But at the same time this means if anyone is using active, everyone is basically using active… no real way to hide.

Exactly. Some thoughts on PvP (the only time there's enemy active sonar):

Real life subs are sunk in a single shot, so it's critical to fire without being fired upon, and therefore to detect without being counter-detected. But in Barotrauma, sniping from the shadows isn't possible because sub weapons have such limited ranges (ignoring shuttle torpedoes). That leaves ambushing from behind a corner as the only advantage a non-pinging sub could have over a pinging sub. But levels don't usually generate with suitable cover, especially such that could hide a large submarine, so this kind of ambushing doesn't work. And even if it did, Baro subs are tough enough to slug it out for tens of seconds at least, so "fire without being fired upon" isn't likely.

Furthermore, the PvP combat mission is a contrived situation: normally subs may be tailing each other, or tailing surface ships for long periods of time, waiting for orders or opportunity to strike, or trying to sneak into a harbor to lay mines, sink ships, or infil/exfill commandos. The situation where "we know there's a hostile sub approaching from this direction in the next few minutes" is… unusual.

So since Barotrauma subs are tanky melee fighters approaching each other head-on, there's little advantage to stealth or silence.

**Comment by Ek-Videogames** [comment]
Rokvach says that in the game monsters and other things are supposed to show up on passive sonar???:
> "the way the passive sonar works in the game is that it picks up objects that are emitting sounds (monsters, devices, characters moving inside the sub, etc)."

Yet it seems like in the game any of these examples regardless of actual 'sound level' are often not visible on passive sonar until they are very close (about 1/4 of the sonar range or less). 

Sounds to me like **the 'echoing' feature is working** _(Weird and not realistic way to go about doing that but whatever I don't care too much either way as long as it works)_ because you get a very small view range from the ship's own sound echoing back - but if we are supposed to be able to detect things based on sounds other than our own sounds and at different sorts of ranges then that does not appear to be happening.

**Comment by HeatedWafflez** [comment]
I believe the game already does this, but I absolutely see where you're coming from. Picking up external noises is fairly difficult so maybe we could have an upgrade for the navigation terminal that increases its sonar sensitivity?
